electroncash/rpa/paycode.py

```diff
@@ -243,15 +243,16 @@ def generate_transaction_from_paycode(wallet, config, amount, rpa_paycode=None,
         # Generate the raw transaction
         raw_tx_string = tx.as_dict()["hex"]
 
-        # Get the TxId for this raw Tx.
-        double_hash_tx = bytearray(sha256(sha256(bytes.fromhex(raw_tx_string))))
-        double_hash_tx.reverse()
-        txid = double_hash_tx.hex()
+        # Get the 'txinid' (double hash of serialized input) for this raw Tx.
+        txin= tx._inputs[0]
+        txin_script = Transaction.input_script(txin,False,sign_schnorr = True)
+        serialized_signed_input = Transaction.serialize_input(txin,txin_script)
+        double_hash_txin = bytearray(sha256(sha256(bytes.fromhex(serialized_signed_input))))
```

Comment:
There is no need to cast this to `bytearray`. It just wastes cycles. Please remove cast.

Reply:
Also: There is no need to do `sha256(sha256(x))`.  Please add `Hash` to your imports from `.bitcoin` and use `Hash`.  `Hash` does the double-sha for you and is more readable...

Reply:
ok. done.